### PR TITLE
make Project an AttributeContainer

### PR DESCRIPTION
**Pull request by pkasprzyk**

### CODE REVIEW

File: neptune/new/internal/backends/neptune_backend_mock.py [hunk 1]
```diff
@@ -105,30 +107,63 @@
...
+        return [Project(str(uuid.uuid4()), self.PROJECT_NAME, self.WORKSPACE_NAME)]
 
     def get_available_workspaces(self) -> List[Workspace]:
-        return [Workspace(str(uuid.uuid4()), "offline")]
+        return [Workspace(str(uuid.uuid4()), self.WORKSPACE_NAME)]
+
+    # pylint: disable=unused-argument
+    # FIXME: stop ignoring container_type on NPT-11117
+    def _create_container(
+        self, container_id: str, container_type: ContainerType, sys_id
```
Comment (Raalsky) [inline neptune/new/internal/backends/neptune_backend_mock.py:147]: Typing for `sys_id`?

File: neptune/new/project.py [hunk 2]
```diff
@@ -30,9 +35,21 @@ class Project:
...
+        op_processor: OperationProcessor,
+        background_job: BackgroundJob,
+        lock: threading.RLock,
+        workspace: str,
+        project_name: str,
+    ):
+        super().__init__(
+            _id, backend, op_processor, background_job, lock, project_id=_id
+        )
+        self._workspace = workspace
```
Comment (Raalsky) [inline neptune/new/project.py:51]: Maybe `project_name` and `workspace` should be shared via `AttributesContainer` with Runs? Only type verifycation for non-null seems to be reasonable.

File: neptune/new/internal/init_run.py [hunk 3]
```diff
@@ -20,8 +20,6 @@
 from platform import node as get_hostname
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/new/internal/init_run.py:20]: > `internal` doesn't mean to be "private"
it kinda should ;)

added backwards-compa layer, though

File: neptune/new/project.py (hunk 2 above)
Comment (pkasprzyk) [reply neptune/new/project.py]: done, I think

**Review by Raalsky — APPROVED** [review]

**Pull request merged**